### ISSUE TITLE
NumPyEigensolver and NumPyMinimumEigensolver Segmentation fault

### ISSUE THREAD

**Issue by adamb56789** [Qiskit/qiskit-terra#7385]
### Environment

- **Qiskit Terra version**: 0.19.0
- **Python version**: 3.7.10
- **Operating system**: Ubuntu 20.04.3 running as WSL 2

### What is happening?

Using the NumPyEigensolver for calculating ground state energies causes segmentation fault. The problem occurred with my own problem, as well as when running the example code here https://github.com/Qiskit/qiskit-nature/blob/main/docs/tutorials/03_ground_state_solvers.ipynb. The VQE solver works correctly.

### How can we reproduce the issue?

Run this notebook https://github.com/Qiskit/qiskit-nature/blob/main/docs/tutorials/03_ground_state_solvers.ipynb, segfault occurs at [6] ("We can compare the VQE results to the NumPy exact solver and see that they match.").

### What should happen?

There should be no segfault and it should print the correct results, as shown in https://github.com/Qiskit/qiskit-nature/blob/main/docs/tutorials/03_ground_state_solvers.ipynb.

**Comment by jakelishman** [comment]
Please could you show the output of
```python
>>> import numpy
>>> numpy.show_config()
```
for this one?

**Comment by adamb56789** [comment]
```python
>>> import numpy
>>> numpy.show_config()
blas_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/home/username/miniconda3/envs/hp/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/home/username/miniconda3/envs/hp/include']
blas_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/home/username/miniconda3/envs/hp/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/home/username/miniconda3/envs/hp/include']
lapack_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/home/username/miniconda3/envs/hp/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/home/username/miniconda3/envs/hp/include']
lapack_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/home/username/miniconda3/envs/hp/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/home/username/miniconda3/envs/hp/include']
Supported SIMD extensions in this NumPy install:
    baseline = SSE,SSE2,SSE3
    found = SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,FMA3,AVX2
    not found = AVX512F,AVX512CD,AVX512_KNL,AVX512_KNM,AVX512_SKX,AVX512_CNL
```

**Comment by Cryoris** [comment]
This is probably not something on Qiskit's end but rather NumPy. Can you run `numpy.linalg.eigvals` for a matrix of the same size used in the tutorials; `2^(number of qubits in the hamiltonian`? If it's not that it might be the Hamiltonian to matrix conversion?

**Comment by adamb56789** [comment]
I was able to fix this by updating numpy from 1.21.2 to the latest at the time, 1.21.5. I have no idea why that worked. Thanks for your suggestions.
